Sources/BraintreePayPal/BTPayPalDriver.m

```diff
@@ -152,6 +152,14 @@ - (void)requestPayPalCheckout:(BTPayPalRequest *)request
             if (request.amount != nil) {
                 parameters[@"amount"] = request.amount;
             }
+            if (request.requestBillingAgreement) {
+                parameters[@"request_billing_agreement"] = @(request.requestBillingAgreement);
+                if (request.billingAgreementDescription.length > 0) {
+                    NSMutableDictionary *billingAgreementDetails = [NSMutableDictionary dictionary];
+                    billingAgreementDetails[@"description"] = request.billingAgreementDescription;
+                    parameters[@"billing_agreement_details"] = billingAgreementDetails;
```

Comment:
This block could be simplified down to
`parameters[@"billing_agreement_details"] = @{@"description": request.billingAgreementDescription};`